amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/dynamodb/DynamoDBLeaseSerializer.java

```diff
@@ -220,6 +228,13 @@ protected AttributeValueUpdate putUpdate(AttributeValue attributeValue) {
             result.put(PENDING_CHECKPOINT_SEQUENCE_KEY, AttributeValueUpdate.builder().action(AttributeAction.DELETE).build());
             result.put(PENDING_CHECKPOINT_SUBSEQUENCE_KEY, AttributeValueUpdate.builder().action(AttributeAction.DELETE).build());
         }
+
+        if (lease.pendingCheckpointState() != null) {
```

Comment:
what happens during the rollout of this feature ? If a host update with pending checkpoint does it ignored if it is being read by the host which does not have 2 phase commit changes on the host.

Reply:
Hosts without the pending checkpoint won't have this when calling the DDB update, so it'll be a NoOp to leases that now have the pending checkpoint.